### PR TITLE
(feat): Add Gitlab e2e pipeline with test cases for Generic Charts

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <uditgaurav@gmail.com>

- This PR contains the Gitlab e2e pipeline with BDD and other test cases for Generic experiments of litmus.
The Different Stages of the pipeline are:
- Setup
- Install
- Deploy
- Generic-experiment
- Infra-experiment
- App-cleanup
- Cleanup

[View Pipeline](https://gitlab.com/uditgaurav1/litmus-2e2-testing/pipelines/108063168)

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  script: 
+    - chmod 755 ./build/gitlab/stages/1-cluster-setup/gcp
+    - ./build/gitlab/stages/1-cluster-setup/gcp
+
+  artifacts:
+    when: always
+    paths:
+      - .kube/
+
+InstallLitmus:
```
Comment (ksatchit) [inline .gitlab-ci.yml]: LitmusInfraSetup

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  - setup
+  - install
+  - deploy
+  - generic-experiment
+  - infra-experiment
+  - push
+  - app-cleanup
+  - cleanup
+
+ClusterSetup:
```
Comment (ksatchit) [inline .gitlab-ci.yml]: Cluster Setup

File: experiments/generic/container-kill.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      shell: kubectl create -f https://hub.litmuschaos.io/api/chaos?file=charts/generic/container-kill/experiment.yaml -n {{ a_ns}}
+
+    - template:
+        src: ../../utils/chaosengine.j2
+        dest: ../../utils/chaosengine.yml
+
+    - name: Create chaos-engine
+      shell: kubectl create -f ../../utils/chaosengine.yml
+
+    - name: Waiting for the chaos complition
```
Comment (ksatchit) [inline experiments/generic/container-kill.yml]: typo

File: experiments/generic/container-kill.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+- hosts: localhost
+  vars:
+    a_ns: litmus
+    a_kind: deployment
+    name: engine-nginx2
+    a_label: run=nginx
+    e_name: container-kill
```
Comment (ksatchit) [inline experiments/generic/container-kill.yml]: identify mandatory vars

File: experiments/generic/pod-cpu-hog.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      until: "'Succeeded' in result.stdout"
+      delay: 10
+      retries: 50
+
+    - name: Checking the verdict of the experiment 
+      shell: kubectl get chaosresult {{ name }}-{{ e_name }} -n {{ a_ns }} -o jsonpath='{.spec.experimentstatus.verdict}'
+      register: cr
+
+    - name: Result of verdict test
+      fail:
```
Comment (ksatchit) [inline experiments/generic/pod-cpu-hog.yml]: does it return with rc != 0

Comment (ksatchit) [reply experiments/generic/pod-cpu-hog.yml]: if experiment fails (verdict fails) - gitlab job also should fail

Comment (uditgaurav) [reply experiments/generic/pod-cpu-hog.yml]: If any tasks of ansible get fail then thee jobs will automatically fail

Comment (ksatchit) [reply experiments/generic/pod-cpu-hog.yml]: cool!

**Pull request merged**